README.adoc

```diff
@@ -1,343 +1,247 @@
-
-// end::spark-build[]
-
-// tag::spark-devdocs[]
-// tag::spark-example[]
-
-== Example Applications
-
-After the jars are built, let's populate a SolrCloud index with tweets (be sure to update the command shown below with your Twitter API credentials):
-
-Start Solr running in Cloud mode and create a collection named “socialdata” partitioned into two shards:
+=== Import jar file via spark-shell
 
 [source]
-bin/solr -c && bin/solr create -c socialdata -shards 2
-
-NOTE: The remaining sections in this document assume Solr is running in cloud mode on port 8983 with embedded ZooKeeper listening on `localhost:9983`.
-
-Also, to ensure you can see tweets as they are indexed in near real-time, you should enable auto soft-commits using Solr’s Config API.
-(tested with 5.4.0, but trunk as of Jan '15 is 6.0.0, which uses a newer set of API endpoints for the Config API and this command will need to be modified)
-Specifically, for this exercise, we’ll commit tweets every 2 seconds.
+cd $SPARK_HOME
+./bin/spark-shell --packages "com.lucidworks.solr:spark-solr:2.0.0"
+                          or 
+./bin/spark-shell --jars spark-solr-2.0.0.jar
 
-[source]
-curl -X POST http://localhost:8983/solr/socialdata/config \
- -d '{"set-property":{"updateHandler.autoSoftCommit.maxTime":"2000"}}'
+=== Connect to your SolrCloud instance
 
-Now, let’s populate Solr with tweets using Spark streaming:
+==== via DataFrame
 
 [source]
-$SPARK_HOME/bin/spark-submit --master $SPARK_MASTER \
- --conf "spark.executor.extraJavaOptions=-Dtwitter4j.oauth.consumerKey=? -Dtwitter4j.oauth.consumerSecret=? -Dtwitter4j.oauth.accessToken=? -Dtwitter4j.oauth.accessTokenSecret=?" \
- --class com.lucidworks.spark.SparkApp \
- ./target/spark-solr-1.0-SNAPSHOT-shaded.jar \
- twitter-to-solr -zkHost localhost:9983 -collection socialdata
+val options = Map(
+  "collection" -> "{solr_collection_name}",
+  "zkhost" -> "{zk_connect_string}"
+)
+val df = sqlContext.read.format("solr")
+  .options(options)
+  .load
 
-Replace `$SPARK_MASTER` with the URL of your Spark master server. If you don’t have access to a Spark cluster, you can run the Spark job in local mode by passing:
+==== via RDD
 
 [source]
---master local[2]
-
-However, when running in local mode, there is no executor, so you’ll need to pass the Twitter credentials in the `spark.driver.extraJavaOptions` parameter instead of `spark.executor.extraJavaOptions`.
+import com.lucidworks.spark.rdd.SolrRDD
+val solrRDD = new SolrRDD(zkHost, collectionName, sc)
 
-Tweets will start flowing into Solr; be sure to let the streaming job run for a few minutes to build up a few thousand tweets in your socialdata collection. You can kill the job using ctrl-C.
+SolrRDD is an RDD of https://lucene.apache.org/solr/5_4_1/solr-solrj/org/apache/solr/common/SolrDocument.html[SolrDocument]
 
-[NOTE]
-====
-The sample command above includes properties for Twitter API credentials which need to be provided by you.
+==== via RDD ([line-through]#Java#)
```

Comment:
Why does this have a line-through? It's strange to have 2 headings that are the same except one word that is struck out, without any explanation for why.